use_cases/eluc/prescriptors/nsga2/nsga2_utils.py

```diff
@@ -59,20 +59,24 @@ def fast_non_dominated_sort(candidates: list):
 
 def calculate_crowding_distance(front):
     """
-    Calculate crowding distance of each candidate in front.
+    Set crowding distance of each candidate in front.
     """
     n_objectives = len(front[0].metrics)
-    distances = [0 for _ in range(len(front))]
+    for candidate in front:
+        candidate.distance = 0
     for m in range(n_objectives):
-        front.sort(key=lambda candidate: candidate.metrics[m])
-        obj_min = front[0].metrics[m]
-        obj_max = front[-1].metrics[m]
-        distances[0] = float('inf')
-        distances[-1] = float('inf')
-        for i in range(1, len(front) - 1):
-            distances[i] += (front[i+1].metrics[m] - front[i-1].metrics[m]) / (obj_max - obj_min)
-
-    return distances
+        sorted_front = sorted(front, key=lambda candidate: candidate.metrics[m])
+        obj_min = sorted_front[0].metrics[m]
+        obj_max = sorted_front[-1].metrics[m]
+        sorted_front[0].distance = float('inf')
+        sorted_front[-1].distance = float('inf')
+        for i in range(1, len(sorted_front) - 1):
+            if obj_max != obj_min:
+                dist = sorted_front[i+1].metrics[m] - sorted_front[i-1].metrics[m]
+                sorted_front[i].distance += dist / (obj_max - obj_min)
```

Comment:
Set distance rather than creating distance list so that we ensure we are setting the correct distance to the correct value